src/Analyzer.elm

```diff
@@ -0,0 +1,199 @@
+module Analyzer exposing (CalledFunction(..), Find(..), functionCalls)
+
+import Comment exposing (Comment, CommentType(..))
+import Elm.Syntax.Declaration as Declaration exposing (Declaration)
+import Elm.Syntax.Expression as Expression exposing (Expression)
+import Elm.Syntax.ModuleName exposing (ModuleName)
+import Elm.Syntax.Node as Node exposing (Node)
+import Elm.Syntax.Range exposing (Range)
+import Review.ModuleNameLookupTable as LookupTable exposing (ModuleNameLookupTable)
+import Review.Rule as Rule exposing (Error, Rule)
+
+
+type CalledFunction
+    = AnyFromExternalModule ModuleName
+    | FromExternalModule ModuleName String
+    | FromSameModule String
+
+
+type Find
+    = All
+    | None
+    | Some
+
+
+type alias Context =
+    { isFunctionDeclaration : Bool
+    , calledFromRange : Maybe Range
+    , functionsFound : List ( CalledFunction, Maybe Range )
+    , lookupTable : ModuleNameLookupTable
+    }
+
+
+
+{-
+
+
+   TODO:
+   - add documentation
+     - Basic functions must be explicit
+   - recognize operators
+   - indirect calls
+
+
+-}
+
+
+functionCalls :
+    { calledFrom : Maybe String
+    , findFunctions : List CalledFunction
+    , find : Find
+    , comment : Comment
+    }
```

Comment:
Dear @angelikatyborska and @neenjaw, I know you can't review Elm code, but I would like to have your opinion on this function API.

This is basically `assert_call` copied from the Elixir Analyzer. I made some changes:
- Renamed `called_fn` and `calling_fn` to `findFunctions ` and `calledFrom` because I was always confusing the two. Maybe there are better names?
- Made `calling_fn` (`calledFrom `) into a `Maybe` type, `Nothing` means anywhere in the module and `Just function` only in `function`.
- Made `called_fn` (`findFunctions `) into a list a functions, and we can either find `All`, `None` or `Some`. I did this because it's simpler to write that a function should call several other ones and also it's easier to say a function should call one of some choices, this is done in Elixir with `suppress_if` but it only works for two (it also covers about 80% of uses of `suppress_if` so maybe it doesn't need to be implemented). It also handles `assert_call` and `assert_no_call` Ine one function.
- The tool I'm using `elm-review` makes it trivial to return a `Range`, basically the location in the source code of a mistake (this this case, mostly `calling_fn ` or sometimes the position of a function that shouldn't be called). However, I'm not sure how to use this power effectively. What would be awesome would be to add squiggly lines on the source code. I'm not sure if that's possible.

I'd love to have your opinion on these changes, and ask if you've had other ideas of things that could be smoother with `asset_call` in general. Thank you 💛